db.go

```diff
@@ -99,59 +89,38 @@ func searchCourse(query string, args []interface{}) ([]CoursesDB, error) {
 }
 
 // 各パラメーターに問題がないかを確認し、問題なければ整形したものを返す
-func validateSearchCourseOptions(courseName string, courseNameFilterType string, courseOverview string, courseOverviewFilterType string, filterType string, limit string, offset string) (searchCourseOptions, error) {
+func validateSearchCourseOptions(query CourseQuery) error {
+
 	allowedFilterType := []string{filterTypeAnd, filterTypeOr}
-	if !util.Contains(allowedFilterType, filterType) {
-		return searchCourseOptions{}, fmt.Errorf("filterType error: %s, %+v", filterType, allowedFilterType)
+	emptyQuert := true
+	if !util.Contains(allowedFilterType, query.FilterType) {
+		return fmt.Errorf("FilterType error: %s, %+v", query.FilterType, allowedFilterType)
 	}
-	if courseName != "" && !util.Contains(allowedFilterType, courseNameFilterType) {
-		return searchCourseOptions{}, fmt.Errorf("courseNameFilterType error: %s, %+v", courseNameFilterType, allowedFilterType)
+	if query.CourseName != "" {
+		if !util.Contains(allowedFilterType, query.CourseNameFilterType) {
+			return fmt.Errorf("CourseNameFilterType error: %s, %+v", query.CourseNameFilterType, allowedFilterType)
+		}
+		emptyQuert = false
 	}
-	if courseOverview != "" && !util.Contains(allowedFilterType, courseOverviewFilterType) {
-		return searchCourseOptions{}, fmt.Errorf("courseOverviewFilterType error: %s, %+v", courseOverviewFilterType, allowedFilterType)
+	if query.CourseOverview != "" {
+		if !util.Contains(allowedFilterType, query.CourseOverviewFilterType) {
+			return fmt.Errorf("CourseOverviewFilterType error: %s, %+v", query.CourseOverviewFilterType, allowedFilterType)
+		}
+		emptyQuert = false
 	}
 
 	// どのカラムも検索対象としていなければ検索そのものが実行できないので、不正なリクエストである
-	if courseName == "" && courseOverview == "" {
-		return searchCourseOptions{}, errors.New("course_name and course_overview are empty")
+	if emptyQuert {
+		return errors.New("all query str parameter is empty")
```

Comment:
```suggestion
		return errors.New("all parameter is empty")
```

query と str が曖昧な感じになっている印象があったので、なんとなくで改善案を提案